### PR TITLE
Add Line Length Check

### PR DESCRIPTION
**Pull request by pantor**
Currently checks the line length at 120 characters, but that's open to discuss.

### CODE REVIEW

File: lib/Checks/LineLengthCheck.vala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ */
+
+public class ValaLint.Checks.LineLengthCheck : Check {
+    static int maximum_characters = 120;
```
Comment (davidmhewitt) [inline lib/Checks/LineLengthCheck.vala]: Make this a `const` instead of a static and have the variable name as upper case.

File: lib/Checks/LineLengthCheck.vala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    public override void check (Gee.ArrayList<ParseResult?> parse_result, ref Gee.ArrayList<FormatMistake?> mistake_list) {
+        string input = "";
+        foreach (ParseResult r in parse_result) {
+            input += r.text;
+        }
+
+        int line_counter = 1;
+        foreach (string line in input.split ("\n")) {
+            if (line.length > maximum_characters) {
```
Comment (davidmhewitt) [inline lib/Checks/LineLengthCheck.vala]: I've just pushed a test case that fails because of this. I think you want to use `line.char_count` instead of `line.length`.

One calculates the length of the line in bytes, and the other in characters. One character can have more than one byte ;)

**Comment by jeremypw** [comment]
This will not work with classic Mac endings (CR or \r) but not too worried about that.
It also fails a corner case test involving having a \n character within a long string. 

```
assert_warning (line_length_check, "stdout.printf ('This is a looooooong line with newline in a string - first part under 120\nThis is a second section making the total length over 120\n')");
```
 This  test to passes when it should fail. Again, I do not think that should block this as it is unlikely to occur in practice.
